src/test/java/seedu/exercise/logic/commands/CustomRemoveCommandTest.java

```diff
@@ -15,35 +27,123 @@
             new ReadOnlyResourceBook<>(), new ReadOnlyResourceBook<>(),
-            new UserPrefs(), getDefaultPropertyBook());
+            new UserPrefs());
+
+        assertCommandSuccess(customRemoveCommand, model, expectedMessage, expectedModel);
+        assertEquals(new HashSet<>(), propertyBook.getCustomProperties());
+    }
+
+    @Test
+    public void execute_customPropertyRemovedForOneExercise_success() {
+        CustomRemoveCommand customRemoveCommand = new CustomRemoveCommand(VALID_FULL_NAME_REMARK,
+            Optional.of(INDEX_ONE_BASED_FIRST));
+
+        // Expected result
+        Exercise updatedExercise = new ExerciseBuilder().build();
+        ReadOnlyResourceBook<Exercise> updatedExerciseBook =
+            new ExerciseBookBuilder().withExercise(updatedExercise).withExercise(secondExercise).build();
+        String expectedMessage = String.format(CustomRemoveCommand.MESSAGE_SUCCESS_SINGLE_REMOVED,
+            VALID_FULL_NAME_REMARK, INDEX_ONE_BASED_FIRST.getOneBased());
+        Model expectedModel = new ModelManager(updatedExerciseBook, new ReadOnlyResourceBook<>(),
+            new ReadOnlyResourceBook<>(), new ReadOnlyResourceBook<>(),
+            new UserPrefs());
+        Set<CustomProperty> expectedSet = Set.of(REMARK); // Custom property should not be removed
+
+        assertCommandSuccess(customRemoveCommand, model, expectedMessage, expectedModel);
+        assertEquals(expectedSet, propertyBook.getCustomProperties());
+    }
 
-        assertEquals(commandResult, expectedMessage);
-        assertEquals(model, expectedModel);
+    @Test
+    public void execute_fullNameNotUsed_throwsCommandException() {
+        CustomRemoveCommand customRemoveCommand = new CustomRemoveCommand("Date", Optional.empty());
+        assertThrows(CommandException.class, () -> customRemoveCommand.execute(model));
+    }
+
+    @Test
+    public void execute_invalidExerciseIndexUnfilteredList_failure() {
+        CustomRemoveCommand customRemoveCommand = new CustomRemoveCommand(VALID_FULL_NAME_REMARK,
+            Optional.of(INDEX_VERY_LARGE_NUMBER));
+        assertThrows(CommandException.class, () -> customRemoveCommand.execute(model));
     }
 
+    @Test
+    public void equals() {
+        CustomRemoveCommand removeRating = new CustomRemoveCommand(VALID_FULL_NAME_RATING, Optional.empty());
+        CustomRemoveCommand removeRatingNonEmptyIndex = new CustomRemoveCommand(VALID_FULL_NAME_REMARK,
+            Optional.of(INDEX_ONE_BASED_FIRST));
+        CustomRemoveCommand removeRatingDifferentIndex = new CustomRemoveCommand(VALID_FULL_NAME_REMARK,
+            Optional.of(INDEX_ONE_BASED_SECOND));
+        CustomRemoveCommand removeEndDate = new CustomRemoveCommand(VALID_FULL_NAME_END_DATE, Optional.empty());
+        CustomRemoveCommand anotherRemoveRating = new CustomRemoveCommand(VALID_FULL_NAME_RATING, Optional.empty());
+
+        // Same object -> true
```

Comment:
There is a method in `TestUtil.java` class called `assertCommonEqualsTest` which tests for null value returning false and same object returning true. Can replace some methods here with that so that we adhere to DRY as much as possible

Reply:
Noted and changed.